source/gui.cpp

```diff
@@ -699,29 +699,19 @@ void reshade::runtime::draw_overlay_menu_home()
 					}
 				}
 			}
-			for (auto i = _techniques.begin(); i != _techniques.end(); ++i)
+			if (const auto it = std::find_if_not(_techniques.begin(), _techniques.end(), [](const reshade::technique &a) { return a.enabled || a.toggle_key_data[0] != 0; }); it != _techniques.end())
```

Comment:
This seems unnecessary. Just always do the sort without this `if`. That would also pay the price for sorting when no technique is enabled, but there is little reason for the user to click on "Active to top" then anyway, so it doesn't matter.

Reply:
Sorry I have no idea how to the 2pass sorting to 1pass.